spaceship/lib/spaceship/connect_api/response.rb

```diff
@@ -0,0 +1,50 @@
+require_relative './base'
+require_relative './models/model'
+
+module Spaceship
+  module ConnectAPI
+    class Response
+      attr_reader :body
+      attr_reader :status
+
+      def initialize(body: nil, status: nil)
+        @body = body
+        @status = status
+      end
+
+      def client
+        return Spaceship::ConnectAPI::Base.client
+      end
+
+      def next_url
+        return nil if body.nil?
+        links = body["links"] || {}
+        return links["next"]
+      end
+
+      def next_page
+        url = next_url
+        return nil if url.nil?
+        return client.get(url)
+      end
+
+      def all_pages
```

Comment:
This eager loading of the results feels a little heavy-handed. Do we have a page_size in the response? We could make this lazy by extending `Enumerable` and implementing `#each` which would yield a single data hash.

The one thing that makes me hesitant about this is cognitive overhead of using such an API. Would it be clear to the consumer of whats happening?

Reply:
Ahh yup yup! I was planning on adding a "get X number of pages" or "X number of results" to eventually 🙃 All pages is def not great so I can add that